lib/twitter_api.rb

```diff
@@ -96,15 +96,17 @@ def twitter_credentials_missing?
 
     def link_handles_in(text)
       text.gsub(/(?:^|\s)@\w+/) do |match|
```

Comment:
I'm wondering why we need the `\s` in these, can't we just change to `(?:^|)@\w+` instead? Based on your gist it seems to give the same result. Are we just worried about being too agressive with these?

Reply:
An even bigger question I just realized is how is the test spec passing, since that seems to expect whitespace which wouldn't exist: https://github.com/discourse/discourse/blob/main/spec/lib/twitter_api_spec.rb

```
[2] pry(main)> text
=> "#foo #foobar"
[3] pry(main)> text.gsub(/(?:^|\s)#\w+/) do |match|
[3] pry(main)*   hashtag    = match.strip[1..]  
[3] pry(main)*   "<a href='https://twitter.com/search?q=%23#{hashtag}' target='_blank'>##{hashtag}</a>"
[3] pry(main)* end.strip  
=> "<a href='https://twitter.com/search?q=%23foo' target='_blank'>#foo</a><a href='https://twitter.com/search?q=%23foobar' target='_blank'>#foobar</a>"
```

Reply:
Oh I guess since it matches the start of the line `^` OR `\s` whitespace:

![image](https://user-images.githubusercontent.com/920448/172981840-e7ca08b6-a6b1-45a8-8f0e-1cdbda375a1f.png)

Reply:
Sorry as to the first question
>I'm wondering why we need the \s in these, can't we just change to (?:^|)@\w+ instead? Based on your gist it seems to give the same result. Are we just worried about being too agressive with these e.g. words with @ in the middle?

I am not sure what this is accomplishing, we want to parse these no matter where they are in the passed `text` as far as I can tell.

Reply:
Yes makes sense, maybe we get rid of the `^|\s` altogether? Seems like it will pick up `@blah` from anywhere in the line then:

![image](https://user-images.githubusercontent.com/920448/172982236-aaffa5f4-7e17-475b-8725-0d4b93f2b990.png)

`(?:)@\w+`

Reply:
I guess maybe check git blame to see if there is some obvious reason for them being there, but doesn't make sense to me

Reply:
There's some subtle differences - we would lose spaces entirely that way:

```
[1] pry(main)> text = "#foo #foobar"
=> "#foo #foobar"
[2] pry(main)> text.gsub(/(?:^|\s)#\w+/) do |match|
[2] pry(main)*   puts match
[2] pry(main)* end  
#foo
 #foobar
=> ""
[3] pry(main)> text.gsub(/#\w+/) do |match|
[3] pry(main)*   puts match  
[3] pry(main)* end  
#foo
#foobar
=> " "
[4] pry(main)> text.gsub(/(?:)#\w+/) do |match|
[4] pry(main)*   puts match  
[4] pry(main)* end  
#foo
#foobar
=> " "
[5] pry(main)> text.gsub(/(?:^|)#\w+/) do |match|
[5] pry(main)*   puts match  
[5] pry(main)* end  
#foo
#foobar
=> " "
```

Reply:
Fair enough, if it doesn't make sense to change the regex that's fine, the change you made is great :+1: